### ISSUE TITLE
Add decoder-only Llama conditional query model

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#14]
## Motivation

The current [`ConditionalQueryModel`](https://github.com/mmcdermott/EveryQuery-conditional/blob/main/src/every_query/model/conditional_model.py) uses:

1. A bidirectional ModernBERT encoder over patient events.
2. A separate `torch.nn.TransformerDecoder` over query-answer blocks.
3. Cross-attention from the query decoder to the encoded patient.
4. A custom block-causal mask.

Add a second architecture using a single decoder-only Hugging Face Llama backbone that jointly processes the patient history and conditional query sequence. Preserve the existing encoder–decoder architecture under a more explicit model name so the two approaches can be compared directly.

## Proposed sequence

For a patient with (n) queries, construct one sequence:

```text
[p₁, …, pₘ, c₁, d₁, a₁, c₂, d₂, a₂, …, cₙ, dₙ]
```

where:

- `p₁, …, pₘ` are patient event tokens.
- `c_i` is the queried code.
- `d_i` is the query duration or event-bound representation.
- `a_i` is the teacher-forced binary answer to query (i).
- The final `a_n` does not need to be included because no later query conditions on it.

During training, apply the existing binary answer head to every `d_i` hidden state and train its logit against the corresponding `a_i` target.

## Backbone

Use Hugging Face [`LlamaModel`](https://huggingface.co/docs/transformers/model_doc/llama#transformers.LlamaModel), not `LlamaForCausalLM`.

`LlamaModel` is preferable because:

- We need hidden states at selected `d_i` positions.
- The task uses a custom binary classification head rather than a vocabulary-sized language-model head.
- `inputs_embeds` allows code, duration, answer, token-type, and ontology-aware embeddings to be combined before entering the backbone.

Instantiate the model from a configurable `LlamaConfig` and train it from scratch. At minimum, expose:

- `hidden_size`
- `intermediate_size`
- `num_hidden_layers`
- `num_attention_heads`
- `num_key_value_heads`
- `max_position_embeddings`
- `attention_dropout`

Set `use_cache=False` during training.

## Embeddings

Preserve the existing input representations:

- Patient events and query codes should use the same code embedding table.
- Ontology-mixed embeddings should continue to work through `get_input_embeddings()` / `set_input_embeddings()`.
- Scalar durations should use the existing duration MLP.
- Event-bounded durations should retain the boundary-code embedding and learned boundary marker.
- Previous binary answers should use the existing answer embedding.
- Add learned token-type embeddings so the model can distinguish patient events, query codes, durations, and answers.

The existing query block-position embedding is probably redundant with Llama’s RoPE positions. Remove it initially unless there is a concrete reason to preserve it.

## Attention behavior

Use standard token-level causal attention rather than porting `build_block_causal_mask`. This changes one behavior: `c_i` can no longer attend forward to `d_i`. The prediction is read from `d_i`, which can still attend to both `c_i` and `d_i`, so the complete current query remains available at every prediction point. This makes the combined backbone fully autoregressive and ensures that `d_i` cannot attend to `a_i` or later tokens.

The required invariant is:

```text
d_i may attend to:
    all patient events
    every earlier query and teacher-forced answer
    c_i
    d_i

d_i may not attend to:
    a_i
    any later query or answer
```

This follows directly from the token ordering and Llama’s standard causal mask. For example:

```text
d₁ sees: patient, c₁, d₁
d₂ sees: patient, c₁, d₁, a₁, c₂, d₂
d₃ sees: patient, c₁, d₁, a₁, c₂, d₂, a₂, c₃, d₃
```

Teacher forcing makes `a_i` available to later queries but must never expose it to the `d_i` state used to predict it.

Pass a two-dimensional `attention_mask` to `LlamaModel` containing `1` for real tokens and `0` for padding. Llama should construct the causal component internally.

## Padding and positions

Place the query sequence immediately after each patient’s final real event. Avoid leaving right-padded patient positions between the patient history and query tokens.

Ensure:

```text
max_position_embeddings >= max_patient_tokens + 3 * max_queries
```

Use the time based rope indexes that the dataset class already supplies so rope time is based on hours and timedelta tokens aren't passed to the model.

## Loss and outputs

For every valid query within a query sequence:

1. Gather the final hidden state at `d_i`.
2. Pass it through the existing answer MLP.
3. Compute BCE-with-logits against the binary answer.
4. Mask only padded query positions.

The output interface should remain:

```python
answer_logits: Tensor  # (batch, n_queries)
valid_mask: Tensor      # (batch, n_queries)
```

## Inference semantics

Earlier answers in a query sequence are caller-supplied conditioning values. They should not be replaced with the model’s own predictions during ordinary conditional inference.

For the sequence

```text
[patient events, c₁, d₁, a₁, …, cₙ, dₙ]
```

the logit read from `d_i` estimates:

```text
P(a_i = 1 | patient events, c₁, d₁, a₁, …, c_i, d_i)
```

The final `d_n` logit is the requested conditional prediction. The earlier `d_i` logits provide additional training supervision for the autoregressive factorization. Because causal attention prevents each `d_i` from seeing `a_i` or later tokens, all valid query logits can be computed in one forward pass without label leakage.

Generating unknown answers sequentially and feeding the model’s predictions back as later conditions is a separate sampling capability and is out of scope for this issue.

## Tests

Add tests for the following invariants:

- Changing `a_i` does not change logits for queries (1,\ldots,i).
- Changing any future query or answer does not change an earlier query’s logit.
- The logit at `d_i` from a full-sequence forward pass matches the logit obtained from the prefix ending at `d_i`.
- Changing `a_i` is allowed to affect logits for queries after (i).
- Padded query blocks do not contribute to the loss.
- Patient padding cannot affect any query prediction.
- Output shapes remain `(batch, n_queries)`.
- Ontology-wrapped code embeddings are used for both patient events and query/boundary codes.
- The configured maximum sequence length includes both patient and query tokens.

## Model organization and compatibility

Preserve both architectures:

- Rename the existing `ConditionalQueryModel` to `ConditionalQueryEncoderDecoderModel`.
- Keep its ModernBERT encoder, `torch.nn.TransformerDecoder`, cross-attention, `build_block_causal_mask`, configuration fields, and existing behavior intact.
- Implement the new Llama-based architecture as `ConditionalQueryARModel`.
- Update the training configuration so the desired architecture can be selected explicitly.
- Update existing encoder–decoder configurations to instantiate `ConditionalQueryEncoderDecoderModel`.
- Keep a backward-compatible `ConditionalQueryModel` alias if existing imports, configurations, or checkpoints depend on that name.
- Ensure existing encoder–decoder checkpoints can still be loaded with the renamed model.
- Share output containers and genuinely architecture-independent utilities where useful, while keeping architecture-specific masking and configuration isolated.
- Add separate tests for both model classes.
- Update the documentation to explain the two available architectures and their different attention behavior.
